config.go

```diff
@@ -549,6 +551,61 @@ func (s *Settings) GetSender(ctx context.Context, env Environment) (send.Sender,
 	return send.NewConfiguredMultiSender(senders...), nil
 }
 
+type githubAppAuth struct {
+	AppId      int64
+	privateKey []byte
+}
+
+// getGithubAppAuth returns app id and app private key if it exists.
+func (s *Settings) getGithubAppAuth() *githubAppAuth {
+	if s.AuthConfig.Github == nil || s.AuthConfig.Github.AppId == 0 {
+		return nil
+	}
+
+	key, ok := s.Expansions[githubAppPrivateKey]
+	if !ok || key == "" {
+		return nil
+	}
+
+	return &githubAppAuth{
+		AppId:      s.AuthConfig.Github.AppId,
+		privateKey: []byte(key),
+	}
+}
+
+// GetGithubAppClient creates a client to request the installation id of the given owner/repo and returns a new client with the correct credentials.
+func (s *Settings) GetGithubAppClient(ctx context.Context, owner, repo string, retry utility.HTTPRetryFunction, delay utility.HTTPDelayFunction) (*http.Client, error) {
+	authFields := s.getGithubAppAuth()
+	if authFields == nil {
+		return nil, errors.New("Github app settings are not set")
+	}
+	httpClient := utility.GetHTTPClient()
+	itr, err := ghinstallation.NewAppsTransport(httpClient.Transport, authFields.AppId, authFields.privateKey)
+	if err != nil {
+		return nil, err
+	}
+	httpClient.Transport = itr
+	defer utility.PutHTTPClient(httpClient)
```

Comment:
i had a question for you @Kimchelly 
when I'm using just 1 client like I'm doing here, I thought I should delete this defer because we're going to be returning the client

but when I deleted that line, I started getting auth errors and nothing worked. It works in the current state that it's in but I'm not super sure why